python/beeai_framework/tools/storage.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2025 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+import os
+import shutil
+from abc import abstractmethod
+
+
+class PythonFile:
+    def __init__(self, filename: str, id: str, python_id: str) -> None:
+        self._id = id
+        self._python_id = python_id
+        self._filename = filename
+
+    @property
+    def id(self) -> str:
+        return self._id
+
+    @id.setter
+    def id(self, value: str) -> None:
+        self._id = value
+
+    @property
+    def python_id(self) -> str:
+        return self._python_id
+
+    @python_id.setter
+    def python_id(self, value: str) -> None:
+        self._python_id = value
+
+    @property
+    def filename(self) -> str:
+        return self._filename
+
+    @filename.setter
+    def filename(self, value: str) -> None:
+        self._filename = value
+
+
+class Input:
+    def __init__(self, local_working_dir: str, interpreter_working_dir: str, ignored_files: list[str]) -> None:
+        self._local_working_dir = local_working_dir
+        self._interpreter_working_dir = interpreter_working_dir
+        self._ignored_files = ignored_files
+
+    @property
+    def local_working_dir(self) -> str:
+        return self._local_working_dir
+
+    @local_working_dir.setter
+    def local_working_dir(self, v: str) -> None:
+        self._local_working_dir = v
+
+    @property
+    def interpreter_working_dir(self) -> str:
+        return self._interpreter_working_dir
+
+    @interpreter_working_dir.setter
+    def interpreter_working_dir(self, v: str) -> None:
+        self._interpreter_working_eir = v
+
+    @property
+    def ignored_files(self) -> list[str]:
+        return self._ignored_files
+
+    @ignored_files.setter
+    def ignored_files(self, v: list[str]) -> None:
+        self._ignored_files = v
+
+
+class PythonStorage:
```

Comment:
Make all methods async.

Reply:
The `list_files` is called in the `__init__` function of the `PythonTool`.  We probably create a factory function for    `pythonTool` class.

Reply:
Follow the TS implementation where listing files is done asynchronously in `input_schema`.

Reply:
In TS we use `fs.promises`, an asynchronous fileIO library. In python there is no equivalent async fileIO library, there are a few third party ones but I would rather not use third party libs for fileIO. Is there a reason we need the fileIO to be async?

Reply:
We don't need async for working with files. But we will need async to enable our users to implement their storage, which might be external like S3 and thus would require HTTP communication, which ideally should be asynchronous.

Reply:
btw: the class should be abstract